clang/include/clang/Basic/TargetInfo.h

```diff
@@ -747,7 +747,10 @@ class TargetInfo : public virtual TransferrableTargetInfo,
   /// Returns true if the given target supports lock-free atomic
   /// operations at the specified width and alignment.
   virtual bool hasBuiltinAtomic(uint64_t AtomicSizeInBits,
-                                uint64_t AlignmentInBits) const {
+                                uint64_t AlignmentInBits,
+                                bool IsCheriCapability) const {
+    if (IsCheriCapability)
```

Comment:
The RISC-V TargetInfo only sets `MaxAtomicInlineWidth` if A is enabled, probably this should be something like `if (IsCheriCapability && getMaxAtomicInlineWidth() > 0)`. Otherwise we will claim that atomics are always supported.

Reply:
Could you add a test that we still emit a warning in something like the following: https://cheri-compiler-explorer.cl.cam.ac.uk/z/s19hs8

Reply:
That makes sense as a simple way to make it work in all the cases we care about without handling the "have integer but not capability" atomics which is just bad hardware design...

Reply:
Don't we already have coverage for those warnings from upstream? That's just normal non-CHERI stuff.

Reply:
There should definitely be coverage for those warnings somewhere, I wonder if we should just add a CHERI run line to such a test or add one without A extension to the CHERI specific one. I don't think this warning is super important, but at least it will flag when users have the wrong -march= flag and accidentally end up using libcalls.